lightning-invoice/src/payment.rs

```diff
@@ -1413,16 +1475,24 @@ mod tests {
 		) -> Result<Route, LightningError> {
 			Err(LightningError { err: String::new(), action: ErrorAction::IgnoreError })
 		}
+
+		fn build_route_from_hops(
+			&self, _payer: &PublicKey, _hops: &[PublicKey], _route_params: &RouteParameters
+			) -> Result<Route, LightningError> {
+			Err(LightningError { err: String::new(), action: ErrorAction::IgnoreError })
+		}
 	}
 
 	struct TestScorer {
-		expectations: Option<VecDeque<PaymentPath>>,
+		expectations: Option<VecDeque<ExpectedPaymentResult>>,
 	}
 
 	#[derive(Debug)]
-	enum PaymentPath {
-		Failure { path: Vec<RouteHop>, short_channel_id: u64 },
-		Success { path: Vec<RouteHop> },
+	enum ExpectedPaymentResult {
+		PathFailure { path: Vec<RouteHop>, short_channel_id: u64 },
+		PathSuccess { path: Vec<RouteHop> },
+		ProbeFailure { path: Vec<RouteHop>, short_channel_id: u64 },
+		ProbeSuccess { path: Vec<RouteHop> },
```

Comment:
Alright, as this is all part of the `TestScorer`, I now renamed the enum from `ExpectedPaymentResult` to `TestResult` and aligned the cases (i.e., `PaymentSuccess`/`PaymentFailure`/`ProbeSuccess`/`ProbeFailure`). When we're positive that all that should/will stay in `InvoicePayer` (see https://github.com/lightningdevkit/rust-lightning/pull/1567#discussion_r908135291), I should also add another test using the latter two.

Reply:
Not arguing either way, but I think the comment there is about removing the method from `Router` but not the one from `Payer` since we still want `InvoicePayer` to handle the probe events.

Reply:
Sure, we'd still handle it, but if we'd end up just using the sending part from `ChannelManager`, an additional test should probably rather be located there?